lib/worker/form.js

```diff
@@ -14,7 +14,7 @@ const pushDraftToEnketo = ({ Forms }, event) =>
   Forms.getByActeeIdForUpdate(event.acteeId, undefined, Form.DraftVersion)
     .then((maybeForm) => maybeForm.map((form) => {
       // if there was no draft or this form isn't the draft anymore just bail.
-      if ((form.def.id == null) || (form.draftDefId !== form.def.id)) return;
+      if (form.def.id == null || !form.isDraft()) return;
```

Comment:
OK, here's a case where I think we maybe do need to check `form.draftDefId`, not just `form.def.publishedAt`. If the worker starts processing a draft form def, but then realizes that the draft has been replaced with a newer draft, it should stop processing the form def. I think it can only determine that the draft has been replaced by checking `draftDefId`: `publishedAt` indicates whether a form def is a draft, while `draftDefId` indicates whether a form def is the _current_ draft.

But now that I look beyond this line, isn't `Forms.getByActeeIdForUpdate()` being called with `Form.DraftVersion`? I would think that that would guarantee that `form.draftDefId === form.def.id`. 🤔

Reply:
> `publishedAt` indicates whether a form def is a draft, while `draftDefId` indicates whether a form def is the _current_ draft.

Would it make sense to remove the `form.isDraft()` function currently added in this PR, and instead have:

1. `form.def.isDraft()`, and 
2. `form.isCurrentDraft()`

?

Reply:
That makes sense to me! I just approved #1577.